lib/core/auth.js

```diff
@@ -188,44 +193,63 @@ export default class Auth {
   // Token helpers
   // ---------------------------------------------------------------
 
-  getToken (strategy) {
-    const _key = this.options.token.prefix + strategy
+  getToken () {
+    return this._token.get()
+  }
 
-    return this.$storage.getUniversal(_key)
+  setToken (token) {
+    return this._token.set(token)
   }
 
-  setToken (strategy, token) {
-    const _key = this.options.token.prefix + strategy
+  syncToken () {
+    return this._token.sync()
+  }
 
-    return this.$storage.setUniversal(_key, token)
+  resetToken () {
+    return this._token.reset()
   }
 
-  syncToken (strategy) {
-    const _key = this.options.token.prefix + strategy
+  getTokenStatus () {
+    return this._token.getStatus()
+  }
 
-    return this.$storage.syncUniversal(_key)
+  refreshTokenAt () {
+    return this._token.refreshAt()
   }
 
   // ---------------------------------------------------------------
   // Refresh token helpers
   // ---------------------------------------------------------------
 
-  getRefreshToken (strategy) {
-    const _key = this.options.refresh_token.prefix + strategy
+  getRefreshToken () {
+    return this._refreshToken.get()
+  }
+
+  setRefreshToken (refreshToken) {
+    return this._refreshToken.set(refreshToken)
+  }
 
-    return this.$storage.getUniversal(_key)
+  syncRefreshToken () {
+    return this._refreshToken.sync()
   }
 
-  setRefreshToken (strategy, refreshToken) {
-    const _key = this.options.refresh_token.prefix + strategy
+  resetRefreshToken () {
+    return this._refreshToken.reset()
+  }
 
-    return this.$storage.setUniversal(_key, refreshToken)
+  getRefreshTokenStatus () {
+    return this._refreshToken.getStatus()
   }
 
-  syncRefreshToken (strategy) {
-    const _key = this.options.refresh_token.prefix + strategy
+  refreshToken () {
+    if (!this.strategy.refreshController) {
+      return Promise.resolve()
+    }
 
-    return this.$storage.syncUniversal(_key)
+    return Promise.resolve(this.strategy.refreshController.handleRefresh()).catch(error => {
```

Comment:
This can be handled in `RefreshToken`?

Reply:
I think it could be inside **Token** class. Something like `$auth.token.refresh()`. What do you think?

Reply:
Makse sense :)

Reply:
This method could't be moved to **Token** class since it also refreshes the refresh token.